raml/schemas/definitions/rule.json

```diff
@@ -1,22 +1,59 @@
 {
     "$schema": "http://json-schema.org/draft-04/schema#",
-    "type": "object",
-    "properties": {
-      "alg": {
-        "type": "string"
-      },
-      "all":{
-        "type":"array",
-        "items":{
-          "type":"array",
-          "items":{
-            "type":"string"
-          }
+    "definitions": {
+        "_id":         { "type": "string" },
+        "project_id":  { "type": "string" },
+        "alg":         { "type": "string" },
+        "name":        { "type": "string" },
+        "config":      { "type": "object" },
```

Comment:
I would remove these five definitions, and instead just use their `{ "type": "string" }` / etc directly.

We probably do this a lot in other parts of the schema, but it's a bit overboard.